File: impl/app/android/app/src/main/java/com/ak/contexta/ui/components/StatCard.kt
package com.ak.contexta.ui.components

import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import com.ak.contexta.ui.theme.Primary

data class StatCardData(
    val number: String,
    val label: String,
    val sub: String? = null
)

@Composable
fun StatCard(
    stat: StatCardData,
    modifier: Modifier = Modifier
) {
    AppCard(modifier = modifier.fillMaxWidth()) {
        Column(
            horizontalAlignment = Alignment.CenterHorizontally,
            modifier = Modifier.fillMaxWidth()
        ) {
            Text(
                text = stat.number,
                style = MaterialTheme.typography.headlineMedium
            )
            Spacer(modifier = Modifier.height(4.dp))
            Text(
                text = stat.label,
                style = MaterialTheme.typography.bodySmall,
                color = MaterialTheme.colorScheme.onSurfaceVariant
            )
            if (stat.sub != null) {
                Text(
                    text = stat.sub,
                    style = MaterialTheme.typography.labelSmall,
                    color = Primary
                )
            }
        }
    }
}

File: impl/app/flutter/lib/core/components/stat_card.dart
import 'package:flutter/material.dart';

import '../theme/app_colors.dart';
import '../theme/app_dimens.dart';
import '../theme/app_type.dart';
import 'app_card.dart';

class StatCardData {
  const StatCardData({
    required this.number,
    required this.label,
    this.sub,
  });

  final String number;
  final String label;
  final String? sub;
}

/// 统计卡片（基于 AppCard）：居中数字 + 标签 + 可选珊瑚小注。
/// 对照 Kotlin ui/components/StatCard.kt。
class StatCard extends StatelessWidget {
  const StatCard({
    super.key,
    required this.stat,
  });

  final StatCardData stat;

  @override
  Widget build(BuildContext context) {
    return AppCard(
      child: Column(
        children: [
          Text(stat.number, style: AppType.textTheme.headlineMedium),
          const SizedBox(height: AppSpacing.xxs),
          Text(
            stat.label,
            style: AppType.textTheme.bodySmall?.copyWith(
              color: AppColors.bodyText,
            ),
          ),
          if (stat.sub != null) ...[
            Text(
              stat.sub!,
              style: AppType.textTheme.labelSmall?.copyWith(
                color: AppColors.primary,
              ),
            ),
          ],
        ],
      ),
    );
  }
}
